front/pages/api/w/[wId]/use/chats/[cId]/messages/[mId]/index.ts

```diff
@@ -0,0 +1,236 @@
+          offset: { type: "number" },
+          score: { type: "number" },
+        },
+        required: ["text", "offset", "score"],
+      },
+    },
+  },
+  required: [
+    "dataSourceId",
+    "sourceUrl",
+    "documentId",
+    "timestamp",
+    "tags",
+    "score",
+    "chunks",
+  ],
+};
+
+export const chatMessageSchema: JSONSchemaType<ChatMessageType> = {
+  type: "object",
+  properties: {
+    sId: { type: "string" },
+    role: { type: "string" },
+    message: { type: "string", nullable: true },
+    retrievals: {
+      type: "array",
+      items: chatRetrievedDocumentSchema,
+      nullable: true,
+    },
+    query: { type: "string", nullable: true },
+    feedback: { type: "string", nullable: true },
+  },
+  required: ["role"],
+};
+
+export type ChatMessageResponseBody = {
+  message: ChatMessageType;
+};
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<ChatMessageResponseBody | ReturnedAPIErrorType>
+): Promise<void> {
+  const session = await getSession(req, res);
+  const user = await getUserFromSession(session);
+  const auth = await Authenticator.fromSession(
+    session,
+    req.query.wId as string
+  );
+
+  const owner = auth.workspace();
+  if (!owner) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace you're trying to modify was not found.",
+      },
+    });
+  }
+
+  if (!user) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_user_not_found",
+        message: "Could not find the user of the current session.",
+      },
+    });
+  }
+
+  if (!auth.isUser()) {
+    return apiError(req, res, {
+      status_code: 403,
+      api_error: {
+        type: "workspace_auth_error",
+        message: "Only users of the current workspace can retrieve chats.",
+      },
+    });
+  }
+
+  if (!(typeof req.query.cId === "string")) {
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: "Invalid query parameters, `cId` (string) is required.",
+      },
+    });
+  }
+
+  if (!(typeof req.query.mId === "string")) {
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: "Invalid query parameters, `mId` (stringg) is required.",
+      },
+    });
+  }
+
+  const dbChatSession = await ChatSession.findOne({
+    where: {
+      workspaceId: owner.id,
+      sId: req.query.cId,
+    },
+  });
+
+  if (!dbChatSession) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "chat_session_not_found",
+        message:
+          "The chat session for the message you're trying to modify was not found.",
+      },
+    });
+  }
+  const chatSession = {
+    ...dbChatSession,
+    created: dbChatSession.createdAt.getTime(),
+  };
+
```

Comment:
I think we want to add a check here that the chatSession belongs indeed to the user otherwise I can go edit your chatSession by API which is not great

Reply:
ok; adding the check in the sessions api too ( front/pages/api/w/[wId]/use/chats/[cId]/index.ts ) in the process